src/librustc/hir/mod.rs

```diff
@@ -95,6 +95,7 @@ impl fmt::Debug for Lifetime {
 pub struct LifetimeDef {
     pub lifetime: Lifetime,
     pub bounds: HirVec<Lifetime>,
+    pub pure_wrt_drop: bool,
```

Comment:
seems like a comment would be nice here (and below); perhaps including a link to the RFC